src/components/CardList.tsx

```diff
@@ -0,0 +1,40 @@
+// components/CardList.tsx
+import React from 'react';
+import Card from './Card';
+
+interface CardListProps {
+  results: { name: string; url: string }[];
+  onSelect: (name: string) => void;
+  onToggle: (name: string, url: string) => void;
+  selectedItems: Array<{ name: string; url: string }>;
+}
+
+const CardList: React.FC<CardListProps> = ({
+  results,
+  onSelect,
+  onToggle,
+  selectedItems,
+}) => {
+  return (
+    <div className="space-y-4">
+      {results.length > 0 ? (
+        results.map((item) => (
```

Comment:
It's better to get if the item is selected one time and pass this value further
```suggestion
        results.map((item) => {
          const isSelected = selectedItems.some(
            (selected) => selected.name === item.name
          );
          return (
            <Card
              key={item.name}
              name={item.name}
              url={item.url}
              onClick={() => onSelect(item.name)}
              isSelected={isSelected}
              onToggle={() =>
                isSelected ? unselect(item.url) : select(item.url)
              }
            />
          );
        })
```